### PR TITLE
feat: add global payment switch

### PR DESCRIPTION
**Pull request by sidvishnoi**
## Context

- Closes https://github.com/interledger/web-monetization-extension/issues/623
- Closes https://github.com/interledger/web-monetization-extension/issues/471
- Closes https://github.com/interledger/web-monetization-extension/pull/654 (supersedes)

### CODE REVIEW

**Comment by sidvishnoi** [comment]
The toggle doesn't look right here, and it's not clear what the toggle would do. Will add power icon.

File: src/background/services/storage.ts [hunk 1]
```diff
@@ -116,14 +116,6 @@ export class StorageService {
...
   }
 
-  async getWMState(): Promise<boolean> {
-    const { continuousPaymentsEnabled } = await this.get([
-      'continuousPaymentsEnabled',
-    ]);
-
-    return continuousPaymentsEnabled;
-  }
-
```
Comment (sidvishnoi) [inline src/background/services/storage.ts:126]: This was unused.

File: src/background/services/monetization.ts [hunk 2]
```diff
@@ -257,12 +275,27 @@ export class MonetizationService {
   }
 
   async toggleWM() {
```
Comment (raducristianpopa) [inline src/background/services/monetization.ts:277]:
```suggestion
  async toggleContinuousPayments() {
```

File: src/pages/popup/components/TogglePaymentsButton.tsx
Comment (raducristianpopa) [inline src/pages/popup/components/TogglePaymentsButton.tsx:1]: Any particular reason for having two components for this? I think we can just merge them into `TogglePaymentsButton`.

**Review by raducristianpopa — COMMENTED** [review]
As a safety measure, we can check if payments are enabled in the `MonetizationService.pay` as well.

**Review by sidvishnoi — COMMENTED** [review]
test-e2e

**Review by raducristianpopa — APPROVED** [review]

File: src/background/services/monetization.ts [hunk 3]
```diff
@@ -256,13 +274,28 @@ export class MonetizationService {
...
-    const { continuousPaymentsEnabled } = await this.storage.get([
+  async toggleContinuousPayments() {
+    const { continuousPaymentsEnabled, enabled } = await this.storage.get([
       'continuousPaymentsEnabled',
+      'enabled',
     ]);
     const nowEnabled = !continuousPaymentsEnabled;
     await this.storage.set({ continuousPaymentsEnabled: nowEnabled });
-    if (nowEnabled) {
+    if (nowEnabled && enabled) {
```
Comment (DarianM) [inline src/background/services/monetization.ts:284]: qq: why do we have to check both `continuousPaymentsEnabled` and `enabled` here? 🤔 
`enabled` is the bool for the entire extension monetization, right?

Comment (sidvishnoi) [reply src/background/services/monetization.ts:284]: If user has disabled continuous payments from toggle in Settings > Rate, but enabled is true, payments are still not to get through.

**Review by DarianM — APPROVED** [review]

**Pull request merged**